// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.User";
option java_outer_classname = "UserProto";
option objc_class_prefix = "HLW";

package user;

// The Login service definition.
service User {
  // Sends a Login
  rpc Login (UserLoginRequest) returns (UserLoginReply) {}
  // Sends a Register
  rpc Register (UserRegisterRequest) returns (UserRegisterReply) {}
}

// The request message containing the user's accout info.
message UserLoginRequest {
  string username = 1; 
  string password = 2;
}

// The response message containing the result
message UserLoginReply {
  string message = 1;
}


// The request message containing the user's infos.
message UserRegisterRequest {
  string username = 1;
  string password = 2;
  string name = 3;
  string cardnum = 4;
}

// The response message containing the result
message UserRegisterReply {
  string message = 1;
}



//required int32 id = 1;
//required string str = 2;
//optional int32 opt = 3;

